src/core/qgsstringutils.cpp

```diff
@@ -523,34 +522,35 @@ QString QgsStringUtils::insertLinks( const QString &string, bool *foundLinks )
   QString converted = string;
 
   // http://alanstorm.com/url_regex_explained
-  // note - there's more robust implementations available, but we need one which works within the limitation of QRegExp
-  static QRegExp urlRegEx( "(\\b(([\\w-]+://?|www[.])[^\\s()<>]+(?:\\([\\w\\d]+\\)|([^!\"#$%&'()*+,\\-./:;<=>?@[\\\\\\]^_`{|}~\\s]|/))))" );
-  static QRegExp protoRegEx( "^(?:f|ht)tps?://|file://" );
-  static QRegExp emailRegEx( "([\\w._%+-]+@[\\w.-]+\\.[A-Za-z]+)" );
+  // TODO - there's more robust implementations available, but we need one which works within the limitation of QRegularExpression
+  thread_local QRegularExpressionMatchIterator urlRegEx = QRegularExpression( "(\\b(([\\w-]+://?|www[.])[^\\s()<>]+(?:\\([\\w\\d]+\\)|([^!\"#$%&'()*+,\\-./:;<=>?@[\\\\\\]^_`{|}~\\s]|/))))" ).globalMatch( converted );
```

Comment:
This is still wrong -- the regex should be the static variable, NOT the match (also line 528)

Reply:
yes, I kept 3 of your messages open which still need to be addressed